### PR TITLE
feat: handle enums in jsonschema

### LINKED ISSUE #1099: Add `Enum` handling to `jsonschema`
It doesn't seem like this is currently being tested in `jsonschema_test.go`

### PR DESCRIPTION
**Pull request by worstell**
fixes #1099

### CODE REVIEW

**Review by wesbillman — APPROVED** [review]
Nice!!

Does the console work ok with this change as well? I wonder if we can deploy a module with enums and see how the editor works in the enum case. :)

File: backend/schema/jsonschema.go [hunk 1]
```diff
@@ -19,42 +19,38 @@ func DataToJSONSchema(sch *Schema, ref Ref) (*jsonschema.Schema, error) {
...
+				root.Definitions[r.String()] = jsonschema.SchemaOrBool{TypeObject: nodeToJSSchema(n, refs)}
 			}
-			data = monomorphisedData
-
-			ref := fmt.Sprintf("%s.%s", r.Module, refName(r))
-			root.Definitions[ref] = jsonschema.SchemaOrBool{TypeObject: nodeToJSSchema(sch, data, refs)}
-		} else {
-			root.Definitions[r.String()] = jsonschema.SchemaOrBool{TypeObject: nodeToJSSchema(sch, data, refs)}
+		case *Enum:
+			root.Definitions[r.String()] = jsonschema.SchemaOrBool{TypeObject: nodeToJSSchema(n, refs)}
```
Comment (wesbillman) [inline backend/schema/jsonschema.go:47]: Is `Enum` the only `ref` we need to special case here? What makes enum special here? (mostly just so I understand better)

Comment (worstell) [reply backend/schema/jsonschema.go:47]: this logic is essentially universal: 
```
if len(r.TypeParameters) > 0 {
  monomorphisedData, err := n.Monomorphise(r)
  if err != nil {
    return nil, err
  }

  ref := fmt.Sprintf("%s.%s", r.Module, refName(r))
  root.Definitions[ref] = jsonschema.SchemaOrBool{TypeObject: nodeToJSSchema(monomorphisedData, refs)}
} else {
  root.Definitions[r.String()] = jsonschema.SchemaOrBool{TypeObject: nodeToJSSchema(n, refs)}
}
```

but `len(r.TypeParameters) > 0` will only evaluate true when the type is `Data`. this was just my attempt at making this section a bit more elegent — `Monomorphise` is a member function on the `Data` class, so without this switch we'd need to do a type assertion that the resolved type is `Data` to perform this call: `n.Monomorphise(r)`.

the switch also explicitly fails for unhandled Decl types which is nice (*Config, *Database, *Secret, *Verb)

**Review by alecthomas — APPROVED** [review]
Nice

**Pull request merged**